src/frequenz/sdk/actor/power_distributing/result.py

```diff
@@ -0,0 +1,111 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Results from PowerDistributingActor."""
+
+from abc import ABC
+from typing import Set
+
+from .request import Request
+
+
+class Result(ABC):
+    """Base class for the power distributor result."""
+
+    def __init__(self, request: Request) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial request
+        """
+        self.request: Request = request
+
+
+class Success(Result):
+    """Send if setting power for all batteries succeed."""
+
+    def __init__(
+        self, request: Request, succeed_power: int, outside_bound: int
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial request
+            succeed_power: How much power out of requested were successfully set.
+            outside_bound: How much power was ignored, because was outside power
+                bound. If Request.adjust_power = False, then this value is always 0.
+        """
+        super().__init__(request)
+        self.succeed_power: int = succeed_power
+        self.outside_bound: int = outside_bound
+
+
+class AnyFail(Result):
+    """Send if any request to set power failed by microgrid side."""
+
+    # It is very simple class with only data so it should be ok to disable pylint.
+    # All these results should be dataclass but in python < 3.10 it is risky
+    # to derive after dataclass.
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        request: Request,
+        succeed_power: int,
+        failed_power: int,
+        failed_batteries: Set[int],
+        outside_bound: int,
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            request: Initial_request
+            succeed_power: How much power was successfully set.
+            failed_power: How much power failed.
+            failed_batteries: What batteries failed.
+            outside_bound: How much power was ignored, because was outside power
+                bound. If Request.adjust_power = False, then this value is always 0.
```

Comment:
Maybe call the field `excess_power`,  with description something like: "part of the requested power that could not be fulfilled, because it was outside available power bounds."?

and maybe you want to say `==` instead of a single =.

Reply:
Thanks! Done